### ISSUE TITLE
TensorboardLogger doesn't log on every step

### ISSUE THREAD

**Issue by EricWiener** [PyTorchLightning/pytorch-lightning#11496]
## 🐛 Bug
The tensorboard logger doesn't log for every step when both validation and training are writing to the same metric key (might also occur in other situations - not positive).

### To Reproduce
Notebook [here](https://colab.research.google.com/drive/1fxbMLUehQr6VmGMz0rmRosj60jbZQ6M-?usp=sharing)

Note that in the example there are no logs between steps 3.124k and 20k:
<img width="580" alt="CleanShot 2022-01-15 at 09 19 25@2x" src="https://user-images.githubusercontent.com/18071029/149624941-583f8e56-8d7e-44b9-8d69-94848face9a9.png">
<img width="555" alt="CleanShot 2022-01-15 at 09 19 20@2x" src="https://user-images.githubusercontent.com/18071029/149624943-f712b925-4989-49e6-ad5b-a65361b5c37f.png">

### Expected behavior
The metrics should be logged on every step.

### Environment
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           11.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.0+cu111
	- pytorch-lightning: 1.5.8
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.12
	- version:           # 1 SMP Tue Dec 7 09:58:10 PST 2021
	
### Additional context
N/A

cc @awaelchli @edward-io @rohitgr7

**Comment by awaelchli** [comment]
I'm pretty sure this is expected behavior as of right now. The TB summary writer expects a global_step counter that increases monotonically for all logged metrics, and Lightning increases this counter at every optimization step, but not during validation (intentionally). 

You can provide your own counter for the metrics plot: 

```py
class CPUMonitor(Callback):
    def __init__(self):
        self.counter = 0

    def setup(
        self,
        trainer: Trainer,
        pl_module: LightningModule,
        stage: Optional[str] = None,
    ) -> None:
        if not trainer.logger:
            raise MisconfigurationException(
                "Cannot use CPUMonitor callback with Trainer that has no logger."
            )

    def _log_cpu_metric_dict(
        self, hook: str, trainer: LightningModule
    ) -> None:
        """Gets CPU metrics.
        
        Args:
            hook (str): the hook that this was called from (ex. 'on_train_batch_start')
        """
        metrics = {
            f"cpu_metrics/vm_percent/{hook}": torch.tensor(psutil.virtual_memory().percent, dtype=torch.float32),
        }
        for k, v in metrics.items():
            trainer.logger.experiment.add_scalar(k, v, global_step=self.counter)
        self.counter += 1

    def on_train_batch_start(
        self,
        trainer: Trainer,
        pl_module: LightningModule,
        batch: Any,
        batch_idx: int,
        unused: Optional[int] = 0,
    ) -> None:
        self._log_cpu_metric_dict("on_batch_start", pl_module)

    def on_validation_batch_start(
        self,
        trainer: Trainer,
        pl_module: LightningModule,
        batch: Any,
        batch_idx: int,
        unused: Optional[int] = 0,
    ) -> None:
        self._log_cpu_metric_dict("on_batch_start", trainer)
```

**Comment by EricWiener** [comment]
Awesome thank you @awaelchli !

**Comment by awaelchli** [comment]
You are welcome. I only had a very quick shot at this and tested very quickly, please check if it really does what you wanted 😅
